test/t8_geometry/t8_geometry_implementations/t8_gtest_geometry_occ.cxx

```diff
@@ -364,3 +366,105 @@ TEST (t8_gtest_geometry_occ, jacobian)
+    T8_ASSERT (0 <= eclass && eclass < T8_ECLASS_COUNT);
+    Handle_Geom_Surface occ_surface;
+    TColgp_Array2OfPnt point_array (1, 2, 1, 2);
+    TopoDS_Shape shape;
+
+    /*  x--> u-parameter
+    *  |
+    *  v v-parameter
+    *
+    *     point_array  1               2
+    *
+    *         1        -----------------
+    *                  |               |
+    *                  |               |
+    *                  | plane surface |
+    *                  |               |
+    *                  |               |
+    *         2        -----------------
+    */
+
+    point_array (1, 1) = gp_Pnt (0.0, 1.0, 0.0);
+    point_array (2, 1) = gp_Pnt (1.0, 1.0, 0.0);
+
+    point_array (1, 2) = gp_Pnt (0.0, 0.0, 0.0);
+    point_array (2, 2) = gp_Pnt (1.0, 0.0, 0.0);
+
+    occ_surface = GeomAPI_PointsToBSplineSurface (point_array).Surface ();
+    shape = BRepBuilderAPI_MakeFace (occ_surface, 1e-6).Face ();
+
+    geometry_occ = new t8_geometry_occ (2, shape, "occ surface dim=2");
+
+    t8_cmesh_init (&cmesh);
+    t8_cmesh_set_tree_class (cmesh, 0, eclass);
+  }
+  void
+  TearDown () override
+  {
+    t8_cmesh_destroy (&cmesh);
+  }
+  t8_cmesh_t cmesh;
+  t8_eclass_t eclass;
+  t8_geometry_occ *geometry_occ;
+  /* The arrays prescribe the linkage of the element. The face of the element is linked and all edges are not */
+  int faces[1] = { 1 };
+  int edges[8] = { 0 };
+};
+
+TEST_P (class_2d_element_linear_occ_surface, t8_check_2d_element_linear_occ_surface)
+{
+  double vertices_quad[12] = { 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0 };
+  double vertices_tri[9] = { 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 1.0, 0.0 };
+
+  t8_cmesh_set_tree_vertices (cmesh, 0, (eclass == T8_ECLASS_QUAD ? vertices_quad : vertices_tri),
+                              t8_eclass_num_vertices[eclass]);
+
+  /* Parameters of the element in u0, v0, u1, v1... in order of the element vertices */
+  double params_quad[8] = { 0, 1, 1, 1, 0, 0, 1, 0 };
+  double params_tri[6] = { 0, 1, 1, 1, 1, 0 };
+
+  /* Passing of the attributes to the element */
+  t8_cmesh_set_attribute (cmesh, 0, t8_get_package_id (), T8_CMESH_OCC_FACE_ATTRIBUTE_KEY, faces, 1 * sizeof (int), 0);
+  t8_cmesh_set_attribute (cmesh, 0, t8_get_package_id (), T8_CMESH_OCC_EDGE_ATTRIBUTE_KEY, edges,
+                          2 * t8_eclass_num_vertices[eclass] * sizeof (int), 0);
+  t8_cmesh_set_attribute (cmesh, 0, t8_get_package_id (), T8_CMESH_OCC_FACE_PARAMETERS_ATTRIBUTE_KEY,
+                          (eclass == T8_ECLASS_QUAD ? params_quad : params_tri),
+                          2 * t8_eclass_num_vertices[eclass] * sizeof (double), 0);
+
+  /* Register the geometry */
+  t8_cmesh_register_geometry (cmesh, geometry_occ);
+  /* Commit the cmesh */
+  t8_cmesh_commit (cmesh, sc_MPI_COMM_WORLD);
+
+  /* First 6 ref_coords for triangle and all 9 ref_coords for quad */
+  double test_ref_coords[27] = { 0.0, 0.0, 0.0, 0.5, 0.0, 0.0, 1.0, 0.0, 0.0, 0.5, 0.5, 0.0, 0.0, 1.0,
```

Comment:
These are the reference coordinates at the element boundaries, right? This is already covered here: [const double t8_forest_vtk_point_to_element_ref_coords[T8_ECLASS_COUNT][T8_FOREST_VTK_QUADRATIC_ELEMENT_MAX_CORNERS][3]](https://github.com/DLR-AMR/t8code/blob/b062c4b828e0264cea3e6210a75eef6b9edf165c/src/t8_forest/t8_forest_vtk.cxx#L127)

However, one probably cannot access this array here. So just this is just for you information.

Reply:
If you ask me, I would randomize these test reference coordinates or at least check oddly chosen reference coordinates.

If you have no time for this add at least a `TODO` here explaining this.